crates/astria-sequencer-relayer/src/relayer/write/conversion.rs

```diff
@@ -92,3 +120,22 @@ pub(super) fn convert(block: SequencerBlock) -> eyre::Result<Converted> {
         },
     })
 }
+
+fn brotli_compressed_bytes(data: &[u8]) -> eyre::Result<Vec<u8>> {
+    use std::io::Write as _;
+    let compression_params = BrotliEncoderParams {
+        quality: 5,
+        size_hint: data.len(),
+        ..Default::default()
+    };
+    let mut output = Vec::new();
+    {
+        let mut compressor =
+            brotli::CompressorWriter::with_params(&mut output, 4096, &compression_params);
+        compressor
+            .write_all(data)
+            .wrap_err("failed compressing data")?;
```

Comment:
You can probably just `compressor.write_all(data)?` and have the function return an `io::Result<Vec<u8>>` because the context msg here is not very useful.